server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -362,6 +362,10 @@ public Query multiPhraseQuery(String field, TokenStream stream, int slop, boolea
         throw new IllegalArgumentException("Can only use phrase queries on text fields - not on [" + name + "] which is of type [" + typeName() + "]");
     }
 
+    public TokenStream tokenize(String field, String text) {
+        throw new IllegalArgumentException("Can only tokenize text on text fields - not on [" + name + "] which is of type [" + typeName() + "]");
+    }
+
```

Comment:
Why do you need this ? We could check the `index_options` and fail `match` parsing if the field is not indexed with positions or check `tokenized` method on the field type ?